app/src/main/java/io/intrepid/contest/screens/conteststatus/ContestStatusPresenter.java

```diff
@@ -23,20 +31,24 @@ public void onViewCreated() {
     }
 
     private void checkContestStatus() {
-        Timber.d("Contest status API call.");
+        Timber.d("Check contest status");
 
         String contestId = persistentSettings.getCurrentContestId().toString();
 
-        Disposable disposable = restApi.getContestStatus(contestId)
-                .compose(subscribeOnIoObserveOnUi())
-                .subscribe(response -> showContestStatusScreen(response), throwable -> {
-                    Timber.d("API error checking contest status: " + throwable.getMessage());
-
-                    // TODO: once API endpoing works, stop showing message and skipping to next screen
-                    view.showMessage(R.string.error_api);
-                    view.showWaitingSubmissionsFragment(3);
+        Disposable timerDisposable = Observable
+                .interval(API_CALL_INITIAL_DELAY, API_CALL_INTERVAL, API_CALL_INTERVAL_UNIT, ioScheduler)
+                .subscribe(success -> {
+                    Timber.d("Contest status API call");
+                    Disposable apiCallDisposable = restApi.getContestStatus(contestId)
+                            .compose(subscribeOnIoObserveOnUi())
+                            .subscribe(response -> showContestStatusScreen(response), throwable -> {
+                                // TODO: once API endpoing works, stop showing message and skipping to next screen
+                                view.showMessage(R.string.error_api);
```

Comment:
This error handling example is exactly what I imagined as a cleaner alternative on another branch of yours. Even though you need access to the context to get the error message from Res directory, your presenter in theory could have passed in the @stringres as a parameter rather than the actual error as a string.